src/components/visit/visitInstance.tsx

```diff
@@ -0,0 +1,241 @@
+import React from 'react'
+import ChevronDownIcon from '@heroicons/react/outline/ChevronDownIcon'
+import { updateUserData, User, Visit } from 'databaseIntigration'
+
+import EditButton from '@/components/visit/Buttons'
+
+type VisitInstanceProps = Visit
+
+interface VisitInstanceState extends User {
+  isEditable: boolean
+  isOpen: boolean
+  isLoaded: boolean
+}
+
+class VisitInstance extends React.Component<
+  VisitInstanceProps,
+  VisitInstanceState
+> {
+  constructor(props: VisitInstanceProps) {
+    super(props)
+    this.state = {
+      isEditable: false,
+      isOpen: false,
+      isLoaded: false,
+      firstName: props.firstName,
+      lastName: props.lastName,
+      dateTime: props.dateTime,
+      petName: props.petName,
+      walkDist: props.walkDist,
+      duration: props.duration,
+      commuteDist: props.commuteDist,
+      commuteMethod: props.commuteMethod,
+      notes: props.notes
+    }
+  }
+
+  render() {
+    return (
+      <div
+        key={this.props.id}
+        className='m-2 flex flex-col space-y-1 rounded-xl bg-gray p-2 drop-shadow-default'
+      >
+        <div className='flex justify-between'>
+          <div className='relative w-full'>
+            <input
+              type='checkbox'
+              checked={this.state.isOpen}
+              readOnly={true}
+              className='peer absolute h-0 w-0 cursor-pointer opacity-0'
+            />
+
+            <ChevronDownIcon
+              className='absolute top-3 right-5 h-6 w-6 cursor-pointer text-primary transition-transform duration-500 peer-checked:rotate-180'
+              onClick={() =>
+                this.setState({ isOpen: !this.state.isOpen, isEditable: false })
+              }
+            />
+            {this.state.isEditable ? (
```

Comment:
This is a pretty big ternary statement which can make the logic of the code harder to read. I would actually write two more components here `<ReadOnlyVisitInstance />` and `<EditableVisitInstance />`. and then have this component delegate between them. This helps keep both components simple by maintaining the single responsibility principal.

For example:

```tsx
const VisitInstance = ({ ... }) => {
    const [editable, setEditable] = useState(false);

    if (editable) return <EditableVisitInstance onSave={() => setEditable(false)} ... />
    return <ReadOnlyVisitInstance onEdit={() => setEditable(true)} ... />
}
```